webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPNodeRepository.scala

```diff
@@ -186,21 +186,21 @@ class WoLDAPNodeRepository(
 
   override def createNode(node: Node, modId: ModificationId, actor: EventActor, reason: Option[String]): IOResult[Node] = {
     val entry = mapper.nodeToEntry(node)
-    for {
+    nodeLibMutex.writeLock(for {
       con <- ldap
       _   <- con.save(entry).chainError(s"Cannot create node with id '${node.id.value}'")
     } yield {
       node
-    }
+    })
   }
 
   override def deleteNode(node: Node, modId: ModificationId, actor: EventActor, reason: Option[String]): IOResult[Node] = {
     val entry = mapper.nodeToEntry(node)
-    for {
+    nodeLibMutex.writeLock(for {
```

Comment:
this seems to block when deleting a node

Reply:
interestingly, this method is not used

Reply:
the problem was actually with dyn group update triggered by the node deletion (I believe)

Reply:
I was not able to reproduce it (adding/deleting nodes)